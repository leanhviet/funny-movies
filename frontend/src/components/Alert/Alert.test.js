// Libs
import React from 'react'
import { shallow } from 'enzyme'

// Components
import { AlertComponent } from './index'

const mockFn = jest.fn()
const props = {
  onClose: mockFn,
  heading: 'Error',
  message: 'All of fields are required',
  variant: 'danger'
}

describe('Alert component', () => {
  const wrapper = shallow(<AlertComponent />)
  const wrapperWithProps = shallow(<AlertComponent {...props} />)

  it('should be defined Alert component', () => {
    expect(wrapper).toBeDefined()
  })

  // Testing Snapshot
  it('should be render snapshot Alert component', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapperWithProps).toMatchSnapshot()
  })

  // // Testing default props
  it('should be render defaultProps Alert component', () => {
    expect(AlertComponent.defaultProps.heading).toEqual('')
    expect(AlertComponent.defaultProps.message).toEqual('')
    expect(AlertComponent.defaultProps.variant).toEqual('')
    expect(AlertComponent.defaultProps.onClose()).toBeUndefined()
  })

  // Testing mock props
  it('should be render mockProps Button component', () => {
    expect(wrapperWithProps.find('Alert').props().variant).toEqual(
      props.variant,
    )
    expect(wrapperWithProps.find('AlertHeading').props().children).toEqual(
      props.heading,
    )
    expect(wrapperWithProps.find('p').props().children).toEqual(
      props.message,
    )
  })

  // Testing for action click
  it('behavior Alert component', () => {
    wrapperWithProps.find('span').simulate('click', {})
    expect(mockFn).toHaveBeenCalledTimes(1)
  })
})
