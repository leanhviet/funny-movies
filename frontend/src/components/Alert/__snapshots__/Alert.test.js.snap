// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component should be render snapshot Alert component 1`] = `
<Alert
  className="alert-cmp"
  closeLabel="Close alert"
  show={true}
  transition={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "appear": false,
        "in": false,
        "mountOnEnter": false,
        "timeout": 300,
        "unmountOnExit": false,
      },
      "render": [Function],
    }
  }
  variant=""
>
  <div>
    <AlertHeading />
    <p />
  </div>
  <span
    className="alert-cmp__closeBtn"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-times"
    />
  </span>
</Alert>
`;

exports[`Alert component should be render snapshot Alert component 2`] = `
<Alert
  className="alert-cmp"
  closeLabel="Close alert"
  show={true}
  transition={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "appear": false,
        "in": false,
        "mountOnEnter": false,
        "timeout": 300,
        "unmountOnExit": false,
      },
      "render": [Function],
    }
  }
  variant="danger"
>
  <div>
    <AlertHeading>
      Error
    </AlertHeading>
    <p>
      All of fields are required
    </p>
  </div>
  <span
    className="alert-cmp__closeBtn"
    onClick={[MockFunction]}
  >
    <i
      aria-hidden="true"
      className="fa fa-times"
    />
  </span>
</Alert>
`;
